webapp/src/js/services/telemetry.js

```diff
@@ -0,0 +1,215 @@
+var moment = require('moment'),
+  uuid = require('uuid/v4');
+
+angular
+  .module('inboxServices')
+  .factory('Telemetry', function($log, $q, $window, DB, Session) {
+    'use strict';
+    'ngInject';
+
+    var queue = $q.resolve();
```

Comment:
I don't understand if there are performance implications to keeping a queue of all telemetry promises for the duration of the browser session.  Hopefully there are not...

Reply:
Yeah I wondered about this too. I think I resolved it in my brain as being OK, but I don't remember what my reasoning was. @garethbowen do you remember why we thought this was OK? And wouldn't, for example, leak memory because it has to endlessly build up a stack trace?

If we're unsure at all IMO we should change tack to being a different form of queuing, though I'm not sure what that would be, since we need to wait until each promise is done, so I think it has to be an endless chain? @alxndrsn what would another implementation here be?

To be super clear, we keep on reassigning `queue` on each call, so we are "dropping the head" from a memory allocation point of view.

Reply:
> To be super clear, we keep on reassigning `queue` on each call, so we are "dropping the head" from a memory allocation point of view.

:+1: I guess we just need to know if `Promise`s keep a reference to their parent.  A naive implementation probably would, but perhaps the implementation is smart enough not to.

Reply:
We ended up discussing this in the development channel but I don't know if we got any closer to an answer. I feel like we went with it as a quick fix for race conditions we were having. I think a different implementation would be easier to follow...